middle_end/flambda2/types/grammar/type_grammar.mli

```diff
@@ -29,6 +29,12 @@ module Block_size : sig
   val inter : t -> t -> t
 end
 
+type relation = private
+  | Is_int of Name.t
+  | Get_tag of Name.t
```

Comment:
That name might be a bit too generic. `int_returning_relation` maybe ?

Reply:
Another possibility is to put them in a naked_immediate submodule that would make it obvious